### ISSUE TITLE
Processor graphql queries can be made more efficient

### ISSUE THREAD

**Issue by cpurta** [subsquid/hydra#151]
TL;DR; The processor when attempting to retrieve substrate events from an indexer will duplicate event names in the graphql query making it less efficient and is leading to NGINX timeout against the Polkadot SubSquid Indexer.

I am currently attempting to process staking rewards/slashes on the Polkadot substrate chain. When running the processor with `DEBUG=* VERBOSE=true` the processor gives the following error message:

```
Failed to connect to the indexer endpoint "https://polkadot.indexer.gc.subsquid.io/v4/graphql" after 1 attempts
```

Digging into this, the query event names used are not unique. Example: 

```graphql
query {
	EVENT: substrate_event(
		where: {
			name: {
				_in: [
					"staking.Bonded"
					"staking.Reward"
					"staking.Slash"
					"staking.Reward"
					"staking.Slash"
				]
			}
			id: { _gt: "0000000000-000000" }
			blockNumber: { _gt: -1, _lte: 99999 }
		}
		limit: 1000
		order_by: { id: asc }
	) {
		id
		name
		method
		params
		indexInBlock
		blockNumber
		blockTimestamp
		extrinsic {
			id
			method
			section
			versionInfo
			signer
			args
			signature
			hash
			tip
		}
	}
}
```

More specifically the `where` input:

```
where: {
	name: {
		_in: [
			"staking.Bonded"
			"staking.Reward"
			"staking.Slash"
			"staking.Reward"
			"staking.Slash"
		]
	}
	id: { _gt: "0000000000-000000" }
	blockNumber: { _gt: -1, _lte: 99999 }
}
```

When running this in my rest client it will hit the NGINX timeout limit thus throwing a 504 Gateway Timeout. I cleaned up the list of event names to be unique in the query and it returned data in approximately 11s. The `where` changed to:

```
where: {
	name: {
		_in: [
			"staking.Bonded"
			"staking.Reward"
			"staking.Slash"
		]
	}
	id: { _gt: "0000000000-000000" }
	blockNumber: { _gt: -1, _lte: 99999 }
}
```

It seems like the event names should be mapped so that they are unique and make the query more efficient.

**Comment by dzhelezov** [comment]
incorporated in `subsquid/squid`
